spark/v3.1/build.gradle

```diff
@@ -17,24 +17,20 @@
  * under the License.
  */
 
+String scalaVersion = System.getProperty("scalaVersion") != null ? System.getProperty("scalaVersion") : System.getProperty("defaultScalaVersion")
+
 def sparkProjects = [
     project(':iceberg-spark:iceberg-spark-3.1_2.12'),
     project(":iceberg-spark:iceberg-spark-extensions-3.1_2.12"),
     project(':iceberg-spark:iceberg-spark-runtime-3.1_2.12')
 ]
 
 configure(sparkProjects) {
-  project.ext {
-    sparkVersion = '3.1.3'
-  }
 
   configurations {
     all {
       resolutionStrategy {
-        force 'com.fasterxml.jackson.module:jackson-module-scala_2.12:2.11.4'
-        force 'com.fasterxml.jackson.module:jackson-module-paranamer:2.11.4'
-        force 'com.fasterxml.jackson.core:jackson-core:2.11.4'
-        force 'com.fasterxml.jackson.core:jackson-databind:2.11.4'
+        force "com.fasterxml.jackson.module:jackson-module-scala_${scalaVersion}:${libs.versions.jackson211.get()}"
```

Comment:
what about forcing the other versions? I think integration tests might fail if those versions aren't aligning

Reply:
PR was opened for so long that I forgot the reason why I did that in the first place, I reverted it back